src/jit/codegenlinear.cpp

```diff
@@ -18,6 +18,92 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "emit.h"
 #include "codegen.h"
 
+/*
+*   Iniatilize spill tracking logic and mark register holding arguments at the beggining of the fucntion (first block)
+*/
+void CodeGen::initializeRegisterSet()
+{
+    unsigned   varNum;
+    LclVarDsc* varDsc;
+
+    /* Initialize the spill tracking logic */
+
+    regSet.rsSpillBeg();
+
+    /* If any arguments live in registers, mark those regs as such */
+
+    for (varNum = 0, varDsc = compiler->lvaTable; varNum < compiler->lvaCount; varNum++, varDsc++)
+    {
+        /* Is this variable a parameter assigned to a register? */
+
+        if (!varDsc->lvIsParam || !varDsc->lvRegister)
+        {
+            continue;
+        }
+
+        /* Is the argument live on entry to the method? */
+
+        if (!VarSetOps::IsMember(compiler, compiler->fgFirstBB->bbLiveIn, varDsc->lvVarIndex))
+        {
+            continue;
+        }
+
+        /* Is this a floating-point argument? */
```

Comment:
It is not clear for me why we skip floating-point arguments here, @CarolEidt maybe we can add a comment about that?